2-Browsers/Week1/homework/ex5-catWalk/index.js

```diff
@@ -19,8 +19,28 @@
 
    https://media1.tenor.com/images/2de63e950fb254920054f9bd081e8157/tenor.gif
 -----------------------------------------------------------------------------*/
+
+const img = document.querySelector('img');
+img.style.left = '0px';
+
 function catWalk() {
-  // TODO complete this function
+  img.style.left = `${parseFloat(img.style.left) + 10}px`;
+
+  if (parseFloat(img.style.left) > window.screen.availWidth) {
+    img.style.left = '0px';
+  }
+
+  if (parseFloat(img.style.left) > window.screen.availWidth / 2) {
```

Comment:
Every time `catWalk()` is called while the cat has crossed the middle of the screen the condition of this `if` statement holds true. It should be executed once only for each left-to-right catwalk. So you could introduce an extra (global) boolean variable, say, `hasDanced`, initially `false`, that you set to `true` when the cat has danced and reset to `false` when resetting the `left` position to zero.

```suggestion
  if (!hasDanced && parseFloat(img.style.left) > window.screen.availWidth / 2) {
```